pandas/core/dtypes/common.py

```diff
@@ -1219,18 +1219,7 @@ def is_bool_dtype(arr_or_dtype) -> bool:
 
     if isinstance(arr_or_dtype, ABCIndex):
         # Allow Index[object] that is all-bools or Index["boolean"]
-        if arr_or_dtype.inferred_type == "boolean":
```

Comment:
Hmm I thought we wanted to still deprecate this in 2.1?

Reply:
un-reverted + green